resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodFactory.scala

```diff
@@ -54,6 +57,9 @@ private[spark] class ExecutorPodFactoryImpl(
       org.apache.spark.internal.config.EXECUTOR_CLASS_PATH)
   private val executorJarsDownloadDir = sparkConf.get(INIT_CONTAINER_JARS_DOWNLOAD_LOCATION)
 
+  private val isKerberosEnabled = sparkConf.get(KUBERNETES_KERBEROS_SUPPORT)
+  private val maybeSimpleAuthentication =
+    if (isKerberosEnabled) Some(s"-D$HADOOP_SECURITY_AUTHENTICATION=simple") else None
```

Comment:
Excellent question.

If this is set to "kerberos", then UserGroupInformation code crashes complaining it cannot read kerberos config files like /etc/krb5.conf. So we want to prevent that by suppressing the code path.

As an alternative, we could create a config map containing /etc/krb5.conf and mount it in the driver and executor pods. But that seems to be an overkill.

Now, there is another question. How can setting it to "simple" allows the driver and executor to access secure HDFS? It works because the driver and executor need only the delegation token to access secure HDFS. i.e. They don't need to sign on to Kerberos on their own.

This is counter-intuitive and hard to explain. I'd love to suggestions to make this part easier to read. Maybe we can call the associated variable like `maybeTokenOnlyAuthentication`.

Reply:
Or at least some comments on why this is set to `simple` would be very helpful.